### PR TITLE
Master documents module

### PR DESCRIPTION
**Pull request by tivisse**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/models/mail_message.py [hunk 1]
```diff
@@ -959,6 +959,8 @@ def write(self, vals):
     def unlink(self):
         # cascade-delete attachments that are directly attached to the message (should only happen
         # for mail.messages that act as parent for a standalone mail.mail record).
+        if not self:
```
Comment (tivisse) [inline addons/mail/models/mail_message.py]: What is the purpose of this ?

File: addons/mail/models/mail_thread.py [hunk 2]
```diff
@@ -314,6 +314,8 @@ def write(self, values):
     def unlink(self):
         """ Override unlink to delete messages and followers. This cannot be
         cascaded, because link is done through (res_model, res_id). """
+        if not self:
```
Comment (tivisse) [inline addons/mail/models/mail_thread.py]: Same here ?

File: addons/mail/static/src/js/document_viewer.js [hunk 3]
```diff
@@ -327,6 +328,18 @@ var DocumentViewer = Widget.extend({
             videoElement.pause();
         }
     },
+    _onSplitPDF: function(e, start, end) {
+//        var self = this;
+//        var start = start;
```
Comment (tivisse) [inline addons/mail/static/src/js/document_viewer.js]: Remove.

File: addons/mail/models/mail_message.py (hunk 1 above)
Comment (ThanhDodeurOdoo) [reply addons/mail/models/mail_message.py]: It prevents recursive loop on unlink (according to raphael it should be implemented everywhere compulsorily)

**Pull request closed without merging**